ast.hpp

```diff
@@ -1786,7 +1791,21 @@ namespace Sass {
   public:
     Type_Selector(ParserState pstate, string n)
     : Simple_Selector(pstate), name_(n)
-    { }
+    {
+      size_t pos = n.find('|');
```

Comment:
There's a lot of added duplication in these nodes. Can we hoist it into a shared parent class.